helm/cert-manager-app/templates/deployment.yaml

```diff
@@ -1,35 +1,53 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ .Values.name }}
-  namespace: {{ .Release.Namespace }}
+  name: {{ template "certManager.name" . }}
+  namespace: {{ .Release.Namespace | quote }}
   labels:
-    app: {{ .Values.name }}
-    giantswarm.io/service-type: "managed"
+    app.kubernetes.io/component: "controller"
+    {{- include "certManager.defaultLabels" . | nindent 4 }}
 spec:
   replicas: 1
   selector:
     matchLabels:
-      app: {{ .Values.name }}
+      app.kubernetes.io/component: "controller"
+      {{- include "certManager.selectorLabels" . | nindent 6 }}
   template:
     metadata:
       labels:
-        app: {{ .Values.name }}
-        giantswarm.io/service-type: "managed"
+        app.kubernetes.io/component: "controller"
+        {{- include "certManager.defaultLabels" . | nindent 8 }}
+      annotations:
+        prometheus.io/path: "/metrics"
+        prometheus.io/scrape: 'true'
+        prometheus.io/port: '9402'
     spec:
-      serviceAccountName: {{ .Values.name }}
+      serviceAccountName: {{ template "certManager.name" . }}
       securityContext:
-        runAsUser: {{ .Values.userID }}
-        runAsGroup: {{ .Values.groupID }}
+        runAsUser: {{ .Values.securityContext.userID }}
+        runAsGroup: {{ .Values.securityContext.groupID }}
+      initContainers:
+        - name: disable-webhook-validation
+          image: "{{ .Values.global.image.registry }}/{{ .Values.image.kubectl.name }}:{{ .Values.image.kubectl.tag }}"
+          imagePullPolicy: {{ .Values.image.kubectl.pullPolicy }}
+          args:
+          - label
+          - namespace
+          - {{ .Release.Namespace | quote }}
+          - certmanager.k8s.io/disable-validation=true
+          - --overwrite=true
```

Comment:
This could be a `post-install` job, however it's probably worth catering to users potentially removing the namespace label in the future.